packages/altive_lints/lib/src/assists/add_macro_template_document_comment.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:analyzer/source/source_range.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+
+/// {@template altive_lints.AddMacroTemplateDocumentComment}
+/// A Dart assist that adds a macro template documentation comment to a class
+/// declaration if it does not already have one.
+///
+/// This assist helps in maintaining consistent documentation across the
+/// codebase by ensuring that each class has a macro template comment, which
+/// can be used for generating documentation or for other tooling purposes.
+///
+/// The macro template comment follows the format:
+///
+/// ```dart
+/// /// {[@]template packageName.className}
+/// ///
+/// /// {[@]endtemplate}
```

Comment:
In the example, using `@endtemplate` as is would be interpreted as the end of the template, so I’ve enclosed it in `[]` to avoid this!